spi/src/main/java/org/opensearch/security/spi/resources/ResourceSharingExtension.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.spi.resources;
+
+/**
+ * This interface should be implemented by all the plugins that define one or more resources and need access control over those resources.
+ *
+ * @opensearch.experimental
+ */
+public interface ResourceSharingExtension {
+
+    /**
+     * Type of the resource
+     * @return a string containing the type of the resource. A qualified class name can be supplied here.
+     */
+    String getResourceType();
+
+    /**
+     * The index where resource is stored
+     * @return the name of the parent index where resource is stored
+     */
+    String getResourceIndex();
+
+    /**
+     * The parser for the resource, which will be used by security plugin to parse the resource
+     * @return the parser for the resource
+     */
+    ResourceParser<? extends Resource> getResourceParser();
```

Comment:
One thing to consider is having a method on this interface to assign a service to this extension point that contains methods to interrogate access for this specific type of sharable resource. Could be an alternative to having a client and instead include everything in the SPI.

Reply:
I think client is a better approach as it separates functionalities. SPI is purely intended for providing interfaces that plugins must implement. Client provides a class with APIs that plugin can utilize to implement Access Control.

Reply:
The advantage of supplying a Service/Client from a method in the extension point here is that a plugin knows that if it is provided then that already means that security is installed and enabled. 

A SPI does provide the extension point, but it can be used like a library as well. I think this should be considered instead of publishing a client that plugins need to depend on as a mandatory dependency.